android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Themes.kt

```diff
@@ -16,16 +16,59 @@
 
 package org.smartregister.fhircore.engine.ui.theme
 
+import androidx.appcompat.app.AppCompatDelegate
 import androidx.compose.material.MaterialTheme
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
 
 /**
  * TODO fix issue with ktfmt formatting annotated high order functions. Current workaround below:
  * lambda in this format content: (@Composable() () -> Unit) to allow spotlessApply
  *
  * To enable app theme set darkTheme = isSystemInDarkTheme
  */
+//@Composable
+//fun AppTheme(darkTheme: Boolean = false, content: (@Composable() () -> Unit)) {
+//  MaterialTheme(colors = if (darkTheme) DarkColors else LightColors, content = content)
+//}
+
+class ThemeViewModel : ViewModel() {
+  var isSystemInDarkTheme by mutableStateOf(false)
+    private set
+
+  fun updateTheme(isDarkTheme: Boolean) {
+    isSystemInDarkTheme = isDarkTheme
+    val theme = if (isSystemInDarkTheme) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO
+    AppCompatDelegate.setDefaultNightMode(theme)
+  }
+}
+
 @Composable
-fun AppTheme(darkTheme: Boolean = false, content: (@Composable() () -> Unit)) {
-  MaterialTheme(colors = if (darkTheme) DarkColors else LightColors, content = content)
+fun AppTheme(content: @Composable () -> Unit) {
+  val themeViewModel: ThemeViewModel = viewModel()
+
+  // Set the app's theme based on the system theme
+  val theme = if (themeViewModel.isSystemInDarkTheme) AppCompatDelegate.MODE_NIGHT_YES else AppCompatDelegate.MODE_NIGHT_NO
+  AppCompatDelegate.setDefaultNightMode(theme)
+
+  CompositionLocalProvider(LocalThemeViewModel provides themeViewModel) {
+    content()
+  }
 }
+
+val LocalThemeViewModel = staticCompositionLocalOf<ThemeViewModel> {
+  error("No ThemeViewModel provided")
+}
+
+@Composable
+fun ChildComposable() {
+  val themeViewModel = LocalThemeViewModel.current
+
+  // Access and observe the system theme state from the ThemeViewModel
+
+  val isSystemInDarkTheme by themeViewModel::isSystemInDarkTheme
+
+  // Use the system theme state in your composables
+  // ...
+}
```

Comment:
How should this be used or is for sample usage? If it is sample then we probably don't need it here unless it is for preview thus would need `@Preview` annotation.